### PR TITLE
MOV-1 As a user, I want to change my password before I sign in.

### PR DESCRIPTION
**Pull request by GramescuDan**
MOV-1 Done

### CODE REVIEW

File: src/main/java/org/loose/fis/mov/controllers/RegistrationController.java [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
 import org.loose.fis.mov.services.UserService;
 
+import javafx.event.ActionEvent;
+import java.io.IOException;
+import java.net.URL;
 import java.util.Objects;
 import java.util.regex.Pattern;
 
-public class RegistrationController {
+public  class RegistrationController {
```
Comment (ionhedes) [inline src/main/java/org/loose/fis/mov/controllers/RegistrationController.java]: Trailing whitespace

File: src/main/java/org/loose/fis/mov/controllers/RegistrationController.java [hunk 2]
```diff
@@ -102,4 +116,19 @@ private void checkFieldsForNull() throws EmptyFieldException {
             throw new EmptyFieldException();
         }
     }
+
+    private Stage stage;
+    private Scene scene;
+    private Parent root;
```
Comment (ionhedes) [inline src/main/java/org/loose/fis/mov/controllers/RegistrationController.java]: Can't these variables be local to `switchToPopUp()`?

> `private Stage stage;`
> `private Scene scene;`
> `private Parent root;`

File: src/main/java/org/loose/fis/mov/services/UserService.java [hunk 3]
```diff
@@ -55,4 +57,11 @@ private static MessageDigest getMessageDigest() {
         }
         return md;
     }
+    public static User findUserByEmail(String email) throws UserNotRegisteredException{
```
Comment (ionhedes) [inline src/main/java/org/loose/fis/mov/services/UserService.java]: Please indent method `findUserByEmail(String)`

File: src/main/java/org/loose/fis/mov/controllers/SceneController.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        session.setDebug(true);
+        try {
+            // Create a default MimeMessage object.
+            MimeMessage message = new MimeMessage(session);
+            // Set From: header field of the header.
+            message.setFrom(new InternetAddress("geani.gibilan@gmail.com"));
+            // Set To: header field of the header.
+            message.addRecipient(Message.RecipientType.TO, new InternetAddress(email));
+            // Set Subject: header field
+            message.setSubject("Salut, barosane!Parola ta");
```
Comment (ionhedes) [inline src/main/java/org/loose/fis/mov/controllers/SceneController.java]: Change to "Kingule, ti-a cazut parola"

File: src/main/java/org/loose/fis/mov/controllers/RegistrationController.java (hunk 2 above)
Comment (GramescuDan) [reply src/main/java/org/loose/fis/mov/controllers/RegistrationController.java]: am scapat de ala

File: src/main/java/org/loose/fis/mov/controllers/RegistrationController.java (hunk 1 above)
Comment (GramescuDan) [reply src/main/java/org/loose/fis/mov/controllers/RegistrationController.java]: nu mai gasesti tu dastea muhahaha

File: src/main/java/org/loose/fis/mov/services/UserService.java (hunk 3 above)
Comment (GramescuDan) [reply src/main/java/org/loose/fis/mov/services/UserService.java]: facuta automat ca baietii

File: src/main/java/org/loose/fis/mov/controllers/SceneController.java (hunk 4 above)
Comment (GramescuDan) [reply src/main/java/org/loose/fis/mov/controllers/SceneController.java]: nu ii mai cade! am pus superglu

File: src/main/java/org/loose/fis/mov/controllers/ChangePasswordBeforeLogInController.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import java.io.IOException;
+
+import static org.loose.fis.mov.services.CommService.WordGenerator;
+import static org.loose.fis.mov.services.UserService.findUserByEmail;
+
+public  class ChangePasswordBeforeLogInController {
+
+    private Stage stage;
```
Comment (ionhedes) [inline src/main/java/org/loose/fis/mov/controllers/ChangePasswordBeforeLogInController.java]: Make these local as well please

**Review by ionhedes — APPROVED** [review]
Good to go

**Pull request merged**